crates/revm/src/processor.rs

```diff
@@ -248,30 +260,46 @@ where
     ) -> Result<ResultAndState, BlockExecutionError> {
         // Fill revm structure.
         #[cfg(not(feature = "optimism"))]
-        fill_tx_env(&mut self.evm.env.tx, transaction, sender);
+        fill_tx_env(&mut self.evm.as_mut().unwrap().context.evm.env.tx, transaction, sender);
 
         #[cfg(feature = "optimism")]
         {
             let mut envelope_buf = Vec::with_capacity(transaction.length_without_header());
             transaction.encode_enveloped(&mut envelope_buf);
-            fill_op_tx_env(&mut self.evm.env.tx, transaction, sender, envelope_buf.into());
+            fill_op_tx_env(
+                &mut self.evm.as_mut().unwrap().context.evm.env.tx,
+                transaction,
+                sender,
+                envelope_buf.into(),
+            );
         }
 
         let hash = transaction.hash();
-        let out = if self.stack.should_inspect(&self.evm.env, hash) {
+        let should_inspect = self
+            .evm
+            .as_ref()
+            .unwrap()
+            .context
+            .external
+            .should_inspect(self.evm.as_ref().unwrap().context.evm.env.as_ref(), hash);
```

Comment:
I don't love how there's now `self.evm.as_ref/mut().unwrap()` in multiple places now, can we hide this?

Reply:
removed with removal of Option